### DISCUSSION TITLE
How to serialize a union without knowing its type?

### DISCUSSION THREAD

**Discussion by niebayes** [nlohmann/json#4152] [Q&A] [answered]
I have a union `Signal` and I want to dump it using this json library. This union only has POD field types. How can I implement the `to_json` and `from_json` for the union?

**Comment by nlohmann** [comment]
See https://json.nlohmann.me/features/arbitrary_types/

**Reply by niebayes** [reply]
Yes, I have read about it before I post this question. But my problem is that I only have a union which can be int32, int64, double64, or any other POD type. I don't know which type the union exactly is.

**Reply by nlohmann** [reply]
Then how should the library know? 🤔

**Reply by niebayes** [reply]
I am a newbee to cpp. Can I cast the union to bytes and serialize the bytes using this library? I can ingest the bytes later on at which moment I would know the exact type of the bytes.

**Reply by nlohmann** [reply, marked as answer]
If you can express this in code, you can implement this in the to_json function. I don't think this is a problem specific to this library.

**Reply by niebayes** [reply]
Thank you for your reply!
